dot/network/transaction.go

```
@@ -54,10 +54,12 @@ func (tm *TransactionMessage) Decode(in []byte) error {
 }
 
 // Hash returns the hash of the TransactionMessage
-func (tm *TransactionMessage) Hash() common.Hash {
-	encMsg, _ := tm.Encode()
-	hash, _ := common.Blake2bHash(encMsg)
-	return hash
+func (tm *TransactionMessage) Hash() (common.Hash, error) {
+	encMsg, err := tm.Encode()
+	if err != nil {
+		return common.Hash{}, nil
```

Should we return an error here?

---

fixed